### PR TITLE
Fix PyTorch 0.3.X issues

### PR DESCRIPTION
**Pull request by GHamrouni**
Fix PyTorch deprecation issues. The code is tested on **PyTorch 0.3.1** in prediction mode, training from scratch and when resuming training from older torch models.

Tested on a custom French dataset (80 epochs) :
[ Num train examples = 210192 ]
[ Num dev examples  = 10570 ]

### CODE REVIEW

**Comment by ajfisch** [comment]
Thanks!

**Comment by ajfisch** [comment]
I think that we might as well just use the explicit `dim` parameter since it works in both versions
